android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -66,25 +63,10 @@ class RemoteSettingsClient(
     /**
      * Updates the local storage with [records] data.
      */
-    suspend fun write(records: List<RemoteSettingsRecord>): RemoteSettingsResult = withContext(Dispatchers.IO) {
+    suspend fun write(records: RemoteSettingsResponse): RemoteSettingsResult = withContext(Dispatchers.IO) {
         try {
-            if (records.isEmpty()) {
-                return@withContext RemoteSettingsResult.DiskFailure(IllegalArgumentException("Cannot write empty list"))
-            }
-            val jsonArray = JSONArray()
-            records.forEach {
-                val recordJson = JSONObject().apply {
-                    put("id", it.id)
-                    put("lastModified", it.lastModified.toString())
-                    put("deleted", it.deleted)
-                    put("attachment", it.attachment)
-                    put("fields", it.fields)
-                }
-                jsonArray.put(recordJson)
-            }
-
             val jsonObject = JSONObject()
-            jsonObject.put("records", jsonArray)
+            jsonObject.put("records", records)
```

Comment:
Here as well we are writing a `RemoteSettingsResponse` using a "record" name. I think we should try to align those things.